azurerm/internal/services/loadbalancer/loadbalancer_resource.go

```diff
@@ -352,13 +386,50 @@ func expandAzureRmLoadBalancerFrontendIpConfigurations(d *pluginsdk.ResourceData
 		}
 
 		if v := data["subnet_id"].(string); v != "" {
+			subnetSet = true
+			properties.PrivateIPAddressVersion = network.IPVersionIPv4
+			if v := data["private_ip_address_version"].(string); v != "" {
+				properties.PrivateIPAddressVersion = network.IPVersion(v)
+			}
 			properties.Subnet = &network.Subnet{
 				ID: &v,
 			}
 		}
 
 		name := data["name"].(string)
-		zones := azure.ExpandZones(data["zones"].([]interface{}))
+		zones := &[]string{"1", "2"}
+		zonesSet := false
+		// TODO - Remove in 3.0
+		if deprecatedZonesRaw, ok := d.GetOk(fmt.Sprintf("frontend_ip_configuration.%d.zones", index)); ok {
+			zonesSet = true
+			deprecatedZones := azure.ExpandZones(deprecatedZonesRaw.([]interface{}))
+			if deprecatedZones != nil {
+				zones = deprecatedZones
+			}
+		}
+
+		if availabilityZones, ok := d.GetOk(fmt.Sprintf("frontend_ip_configuration.%d.availability_zone", index)); ok {
+			zonesSet = true
+			switch availabilityZones.(string) {
+			case "1", "2", "3":
+				zones = &[]string{availabilityZones.(string)}
+			case "Zone-Redundant":
+				zones = &[]string{"1", "2"}
+			case "No-Zone":
+				zones = &[]string{}
+			}
+		}
+		if !strings.EqualFold(sku, string(network.LoadBalancerSkuNameStandard)) {
+			if zonesSet && len(*zones) > 0 {
+				return nil, fmt.Errorf("Availability Zones are not available on the `Basic` SKU")
+			}
+			zones = &[]string{}
+		} else if !subnetSet {
+			if zonesSet && len(*zones) > 0 {
+				return nil, fmt.Errorf("Networking supports zones only for frontendIpconfigurations which reference a subnet.")
+			}
+			zones = &[]string{}
+		}
 		frontEndConfig := network.FrontendIPConfiguration{
```

Comment:
Should we add a len check here since it is only valid to send all zones or a single zone now? If we detect that two zones have been defined we should return an error saying that `two zones have been specified, only one zone or all three zones can be specified for this resource.` We should also make sure that the passed strings are not duplicates of each other (e.g. block the user from passing "2", "2", "2").

Reply:
> two zones have been specified, only one zone or all three zones can be specified for this resource

About these error, it happens when apply zones to a loadbalancer which reference a publicIp/publicIpPrefix (not a subnet)。
Availability zones should only be applied when use standard sku and reference a subnet. So I added check on sku and subnet to make sure of it, in other cases, it will be set to an empty array. So now the tests all passed.

> We should also make sure that the passed strings are not duplicates of each other (e.g. block the user from passing "2", "2", "2").

I think it's not necessary, because zones's maxItem is one.